pkg/apis/config/v1alpha1/adminqos.go

```diff
@@ -131,6 +131,45 @@ type CPUAdvisorConfig struct {
 	// we will rely on kernel features to ensure that shared_cores pods can suppress and preempt reclaimed_cores pods.
 	// +optional
 	AllowSharedCoresOverlapReclaimedCores *bool `json:"allowSharedCoresOverlapReclaimedCores,omitempty"`
+
+	// optional
+	CPUProvisionConfig *CPUProvisionConfig `json:"cpuProvisionConfig"`
+}
+
+type IndicatorTargetConfiguration struct {
+	Name   string  `json:"name"`
+	Target float64 `json:"target"`
+}
+
+// FirstOrderPIDParams holds parameters for pid controller in rama policy
+type FirstOrderPIDParams struct {
+	Kpp                  float64 `json:"kpp"`
+	Kpn                  float64 `json:"kpn"`
+	Kdp                  float64 `json:"kdp"`
+	Kdn                  float64 `json:"kdn"`
+	AdjustmentUpperBound float64 `json:"adjustmentUpperBound"`
+	AdjustmentLowerBound float64 `json:"adjustmentLowerBound"`
+	DeadbandUpperPct     float64 `json:"deadbandUpperPct"`
+	DeadbandLowerPct     float64 `json:"deadbandLowerPct"`
+}
+
+type PolicyRamaConfiguration struct {
+	PIDParameters                   map[string]FirstOrderPIDParams `json:"PIDParameters,omitempty"`
+	EnableBorwein                   bool                           `json:"enableBorwein,omitempty"`
+	EnableBorweinModelResultFetcher bool                           `json:"enableBorweinModelResultFetcher,omitempty"`
+}
+
+type ControlKnobConstraints struct {
+	RestrictControlKnobMaxUpperGap      map[string]float64 `json:"restrictControlKnobMaxUpperGap,omitempty"`
+	RestrictControlKnobMaxLowerGap      map[string]float64 `json:"restrictControlKnobMaxLowerGap,omitempty"`
+	RestrictControlKnobMaxUpperGapRatio map[string]float64 `json:"restrictControlKnobMaxUpperGapRatio,omitempty"`
+	RestrictControlKnobMaxLowerGapRatio map[string]float64 `json:"restrictControlKnobMaxLowerGapRatio,omitempty"`
+}
+
+type CPUProvisionConfig struct {
+	IndicatorTargets       map[string][]IndicatorTargetConfiguration `json:"indicatorTargets,omitempty"`
```

Comment:
the same as before and it is better use list instead of map and add more comments